### PR TITLE
#46: Added developer guide

### LINKED ISSUE #46: Add developer guide
#### Acceptance Criteria

- Added information on creating custom connectors
- Added information on create an AWS marketplace release

### LINKED ISSUE #47: Setup automatic issue creation for new releases
#### Situation

Once we have a new release in this project, we should also publish the artifact to the AWS marketplace. 

Ideally, we should have a step in our release-droid scripts that automatically publishes artifact to marketplace. Then an auto issue should be created in the repository to make the marketplace release public.

#### 

- Added semi manual setup for AWS marketplace publishing

### PR DESCRIPTION
**Pull request by morazow**
Fixes #46
Fixes #47

### CODE REVIEW

File: doc/developers_guide/developers_guide.md [hunk 1]
```diff
@@ -1,3 +1,61 @@
...
+
+## Creating Custom Connector
+
+AWS Glue Studio allows to create a custom connectors. It can be useful to debug the connector.
+
+To test connector by creating a custom connector, please follow these steps.
+
+### Creating an Assembly Jar
+
+By running `mvn verify` or `mvn package` create an connector artifact. For example, `target/exasol-glue-connector-1.0.0-assembly.jar`.
```
Comment (kaklakariada) [inline doc/developers_guide/developers_guide.md:39]:
```suggestion
By running `mvn verify` or `mvn package` create an connector artifact. For example, `target/exasol-glue-connector-1.0.0-assembly.jar`.
```

File: doc/developers_guide/developers_guide.md [hunk 2]
```diff
@@ -1,3 +1,61 @@
...
+
+### Uploading the Artifact to S3 Bucket
+
+Upload the JAR artifact from previous step into an S3 bucket. For instance, `s3://exasol-artifacts/glue-connector/`.
```
Comment (kaklakariada) [inline doc/developers_guide/developers_guide.md:43]:
```suggestion
Upload the JAR artifact from previous step into an S3 bucket. For instance, `s3://exasol-artifacts/glue-connector/exasol-glue-connector-1.0.0-assembly.jar`.
```

File: doc/developers_guide/developers_guide.md [hunk 3]
```diff
@@ -1,3 +1,61 @@
...
+Go to AWS Glue Studio, and click on "Connectors" &rarr; "Create Custom Connector".
+
+![Exasol AWS Glue Studio Custom Connectors](img/custom_connector.png)
+
+Then select `Spark` as "Connection type" from drop-down list and fill other necessary fields.
+
+- "Browse S3" and select the uploaded JAR artifact from previous step
+- Enter unique and descriptive name for the connector
+- Enter `exasol` as "Class name"
+- And click to the "Create connector" button
```
Comment (kaklakariada) [inline doc/developers_guide/developers_guide.md:56]:
```suggestion
- And click on the "Create connector" button
```

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: release/pushImageToECR.sh [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+set -eo pipefail
+
+if ! command -v xmlstarlet &>/dev/null; then
+	echo "xmlstarlet tool is not available, please install it to continue."
+	exit 1
+fi
+
+TAG=$(mvn --file pom.xml -q -Dexec.executable="echo" -Dexec.args='${project.version}' --non-recursive exec:exec)
+echo "==> Preparing AWS release for tag '$TAG' version"
```
[the remark this answers is not in the document]
Comment (morazow) [reply release/pushImageToECR.sh:11]: Yes, it is unfortunate. I was expecting to have github provided value from actions. 
There is `github.ref_name` but it is whole reference, requires parsing, etc. Used above for now.

File: doc/developers_guide/developers_guide.md (hunk 2 above)
Comment (morazow) [reply doc/developers_guide/developers_guide.md:43]: It does not work at the moment. artifact-reference-checker assumes whole text as artifact.
I opened this issue for now: https://github.com/exasol/artifact-reference-checker-maven-plugin/issues/24

File: doc/developers_guide/developers_guide.md (hunk 1 above)
Comment (morazow) [reply doc/developers_guide/developers_guide.md:39]: What did change here? :/

File: .github/ISSUE_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+---
+title: Publish new version to AWS Marketplace
+assignees:
+labels:
+---
+
+A [new version](https://github.com/exasol/glue-connector/releases) of this project is released.
+
+- [ ] Check that the corresponding Docker image is pushed to the AWS ECR Registry in [Prepare AWS Marketplace Action](workflows/prepare_aws_marketplace_release.yml)
+- [ ] Follow the [instructions](../doc/developers_guide/developers_guide.md#developers-guide) in developer guide to publish the new version to the AWS Marketplace
```
[the remark this answers is not in the document]
Comment (morazow) [reply .github/ISSUE_TEMPLATE.md:10]: Yep, it does not work, fixed

File: doc/changes/changes_1.0.0.md [hunk 6]
```diff
@@ -1,6 +1,20 @@
...
+
+In this the following features are supported:
+
+- Mapping Exasol table or query metadata to AWS Glue Studio Spark schema
+- Reading from an Exasol table or provided Exasol SQL query
+- Pruning of columns from Glue dataframe into Exasol database
+- Pushing of filters from Glue dataframe into Exasol database
+- Writing AWS Glue Studio dataframe into an Exasol table
+
+Please checkout the project user guide for more details.
```
[the remark this answers is not in the document]
Comment (morazow) [reply doc/changes/changes_1.0.0.md:17]: Done

File: doc/developers_guide/developers_guide.md (hunk 1 above)
Comment (kaklakariada) [reply doc/developers_guide/developers_guide.md]: Sorry, I guess my mistake

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**